### ISSUE TITLE
Only "sudo woeusb" works ! NOK "woeubgui" or "sudo woeusbgui"

### ISSUE THREAD

**Issue by willemijns** [slacka/WoeUSB#248]
Hello,

woeusb displays that 

Warning: You are not running WoeUSB as root!
Warning: This might be the reason of the following failure.

ok ! "sudo woeusb" works...

but "woeusbgui" or "sudo woeusbgui" gives:

Mounting source filesystem...
Error: File "/media/woeusb_source_1551702303_5168/sources/install.wim" in source image has exceed the FAT32 Filesystem 4GiB Single File Size Limitation and cannot be installed.

=======> I think you need to ask "sudo" for the GUI *and*  add a "sudo mount" not only a "mount" ;)

**Comment by robinhood2014** [comment]
You're quite right. The woeusb command, which gets called from this front-end, absolutely requires the NTFS file system to be passed if the source image is >4GB.

The critical piece of code we want is line 243 of MainPanel.cpp, which currently reads as follows:
`        PipeManager pipe(std::string("pkexec sh -c 'woeusb --no-color --for-gui --device \"") + iso + "\" \"" + device + "\" 2>&1'");`

I think appending `--tgt-fs NTFS` at the end of this line might fix the problem. The revised line should read:
`        PipeManager pipe(std::string("pkexec sh -c 'woeusb --no-color --for-gui --device \"") + iso + "\" \"" + device + "\" 2>&1 --tgt-fs NTFS'");`

One drawback of this fix is that this would force ALL disk images to be written to an NTFS file system, whereas slacka probably wants to leave that decision up to the end user.

Sadly, when I try to run a subsequent `make` command (following successful calls of `autoreconf --force --install`, `autoconf`, and `configure ----with-wx-config=wx-config-gtk3`), everything goes to hell when the error output displayed in the attached file is generated.
[libwebkit2gtk.txt](https://github.com/slacka/WoeUSB/files/3000045/libwebkit2gtk.txt)

**Comment by JonnyTech** [comment]
I am not at a computer but it should be more like:
```
PipeManager pipe(std::string("pkexec sh -c 'woeusb --no-color --for-gui --device \"") + iso + "\" \"" + device + " --tgt-fs NTFS\" 2>&1'");
```
although you may need another ```std::string()``` enclosing it too.

**Comment by Lin-Buo-Ren** [comment]
Patches that adds an configurable UI are always welcomed.

**Comment by robinhood2014** [comment]
I know, right? Unfortunately, I feel more at home with Visual Basic (although Visual C# is pretty similar), so at least from where I'm sitting, it's not easy to add such an improvement to the GUI using C++ -- although, obviously, it CAN be done; otherwise the whole darn GUI front-end would probably not exist.

I'll need to do some more research on C++ GUI design.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by JonnyTech** [comment]
Added in #253, please check @robinhood2014

**Comment by l4jos** [comment]
Same issue here on Parrot OS, if I open ~/woeusb/src/woeusbgui from terminal.

Error : Error from 'MainPanel.cpp' at line 163
Description: Can't read usb list !

It works if you launch woeusb from menu in the panel!

**Comment by l4jos** [comment]
> Same issue here on Parrot OS, if I open ~/woeusb/src/woeusbgui from terminal.
> 
> Error : Error from 'MainPanel.cpp' at line 163
> Description: Can't read usb list !
> 
> It works if you launch woeusb from menu in the panel!

but not anymore
